src/package_name.rs

```diff
@@ -54,14 +46,34 @@ impl PackageName {
     }
 }
 
-fn is_ident_valid(ident: &str) -> bool {
-    let only_valid_chars = ident
+fn validate_scope(scope: &str) -> anyhow::Result<()> {
+    let only_valid_chars = scope
         .chars()
         .all(|char| char.is_ascii_lowercase() || char.is_ascii_digit() || char == '-');
 
-    let valid_length = ident.len() > 0;
+    ensure!(
+        only_valid_chars,
+        "package scope {} in invalid (scopes can only contain lowercase characters, digits and '-')",
```

Comment:
Is this a typo? Do you mean "package scope {} is invalid"?

Reply:
Oops, not sure how I missed that!